flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/SourceReaderFinishedEvent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.source.hybrid;
+
+import org.apache.flink.api.connector.source.SourceEvent;
+
+/**
+ * A source event sent from the HybridSourceReader to the enumerator to indicate that the current
+ * reader has finished and splits for the next reader can be sent.
+ */
+public class SourceReaderFinishedEvent implements SourceEvent {
+
+    private static final long serialVersionUID = 1L;
+    private final int sourceIndex;
+    private final long checkpointId;
+
+    /**
+     * Constructor.
+     *
+     * @param sourceIndex
+     */
+    public SourceReaderFinishedEvent(int sourceIndex, long checkpointId) {
+        this.sourceIndex = sourceIndex;
+        this.checkpointId = checkpointId;
+    }
+
+    public int sourceIndex() {
+        return sourceIndex;
+    }
+
+    public long getCheckpointId() {
```

Comment:
it seems that `checkpointId` is not used in the code. Do you have plan to use `checkpointId` in the future? if not, we probably can remove all the checkpointId related code in the HybridSourceReader class

Reply:
Leftover from experiment. Will remove.